test/e2e/metrics/swaps.spec.js

```diff
@@ -0,0 +1,504 @@
+          environment_type: 'fullscreen',
+          locale: 'en',
+        });
+
+        assert.equal(reqs[13].event, MetaMetricsEventName.SwapCompleted);
+        assert.equal(Object.keys(reqs[13].properties).length, 30);
+        assert.equal(reqs[13].properties.token_from, 'TESTETH');
+        assert.equal(reqs[13].properties.token_from_amount, '2');
+        assert.equal(reqs[13].properties.token_to, 'DAI');
+        assert.equal(typeof reqs[13].properties.token_to_amount, 'string');
+        assert.equal(reqs[13].properties.slippage, 2);
+        assert.equal(reqs[13].properties.custom_slippage, false);
+        assert.equal(reqs[13].properties.best_quote_source, 'airswapV4');
+        assert.equal(
+          typeof reqs[13].properties.other_quote_selected,
+          'boolean',
+        );
+        assert.equal(
+          typeof reqs[13].properties.other_quote_selected_source,
+          'string',
+        );
+        assert.equal(typeof reqs[13].properties.gas_fees, 'string');
+        assert.equal(typeof reqs[13].properties.estimated_gas, 'string');
+        assert.equal(reqs[13].properties.suggested_gas_price, '30');
+        assert.equal(reqs[13].properties.used_gas_price, '30');
+        assert.equal(reqs[13].properties.is_hardware_wallet, false);
+        assert.equal(reqs[13].properties.stx_enabled, false);
+        assert.equal(reqs[13].properties.current_stx_enabled, false);
+        assert.equal(typeof reqs[13].properties.reg_tx_fee_in_usd, 'number');
+        assert.equal(typeof reqs[13].properties.reg_tx_fee_in_eth, 'number');
+        assert.equal(
+          typeof reqs[13].properties.reg_tx_max_fee_in_usd,
+          'number',
+        );
+        assert.equal(
+          typeof reqs[13].properties.reg_tx_max_fee_in_eth,
+          'number',
+        );
+        assert.equal(reqs[13].properties.token_to_amount_received, '');
+        assert.equal(reqs[13].properties.quote_vs_executionRatio, null);
+        assert.equal(reqs[13].properties.estimated_vs_used_gasRatio, '100%');
+        assert.equal(reqs[13].properties.approval_gas_cost_in_eth, 0);
+        assert.equal(
+          typeof reqs[13].properties.trade_gas_cost_in_eth,
+          'number',
+        );
+        assert.equal(
+          typeof reqs[13].properties.trade_and_approval_gas_cost_in_eth,
+          'number',
+        );
+        assert.equal(reqs[13].properties.category, 'Swaps');
+        assert.equal(reqs[13].properties.locale, 'en');
+        assert.equal(reqs[13].properties.chain_id, '0x539');
+        assert.equal(reqs[13].properties.environment_type, 'background');
+
+        assert.equal(reqs[14].event, MetaMetricsEventName.SwapCompleted);
+        assert.deepStrictEqual(reqs[14].properties, {
+          category: MetaMetricsEventCategory.Swaps,
+          chain_id: toHex(1337),
+          environment_type: 'background',
+          locale: 'en',
+        });
+
+        assert.equal(reqs[15].event, MetaMetricsEventName.ExitedSwaps);
```

Comment:
Optional: can we have some util func to collect the assert based on `MetaMetricsEventName`?

Reply:
Just to double check I understand it correctly, are you suggesting this change?

```js
// before
assert.equal(reqs[15].event, MetaMetricsEventName.ExitedSwaps);

// after
assertRequestEvent(reqs[15], MetaMetricsEventName.ExitedSwaps);
// ... inside `helpers.js` ...
const assertRequestEvent = (request, expectedEvent) => assertEqual(request.event, expectedEvent);
```

Not 100% sure if there's value here because the assertion is a one-liner already.

Reply:
I was more thinking like this: 
```
const valueSet = [12, '2', false, 2, false...]
const comparedProperties = reqs[15].properties;
assertEvents(comparedProperties, valueSet);

// you can reuse this for all the events
function assertEvents(properties, value) {
 Object.keys(properties).forEach((key, index) => {
  assert.equal(properties[key] , valueSet[index]);
  });
}
```

Reply:
Ah, I got it, Danica. Thanks for the suggestion that would make the test smaller. I'd still rather explicitly show the properties' names because it makes them more explicit for us without needing to add logs.